Sources/PersonalMessage.php

```diff
@@ -1083,29 +1083,7 @@ function prepareMessageContext($type = 'subject', $reset = false)
 	// Any custom profile fields?
 	if (!empty($memberContext[$message['id_member_from']]['custom_fields']))
 		foreach ($memberContext[$message['id_member_from']]['custom_fields'] as $custom)
-			switch ($custom['placement'])
-			{
-				case 1:
-					$output['custom_fields']['icons'][] = $custom;
-					break;
-				case 2:
-					$output['custom_fields']['above_signature'][] = $custom;
-					break;
-				case 3:
-					$output['custom_fields']['below_signature'][] = $custom;
-					break;
-				case 4:
-					$output['custom_fields']['below_avatar'][] = $custom;
-					break;
-				case 5:
-					$output['custom_fields']['above_member'][] = $custom;
-					break;
-				case 6:
-					$output['custom_fields']['bottom_poster'][] = $custom;
-					break;
-				default:
-					$output['custom_fields']['standard'][] = $custom;
-			}
+			$output['custom_fields'][$context['cust_profile_fields_placement'][$custom['placement']]][] = $custom;
```

Comment:
Would be nice to place a comment here with the position how exists otherwise this place is such generic,
that this information is hard to see.

Reply:
Thats kinda the point, to be generic. I did it like this to have a single point where custom fields are defined, makes it easier for anyone to add more.

Reply:
Sure and it's fine but the existing point are not visible anymore or
this somewhere already define?

Reply:
On Sources/Load.php https://github.com/SimpleMachines/SMF2.1/blob/c939cab292f53d2de9058fde91c8f5493692aa5f/Sources/Load.php#L324